### PR TITLE
Add 'HasAttribute' filter rule to citations, repositories and sources

### PR DESCRIPTION
**Pull request by Mattkmmr**
Fixes [#0009845](https://gramps-project.org/bugs/view.php?id=9845): Add a filter on attribute for Sources

### CODE REVIEW

File: gramps/gen/filters/rules/citation/_hasattribute.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from .._hasattributebase import HasAttributeBase
+
+
+# -------------------------------------------------------------------------
+#
+# HasAttribute
+#
+# -------------------------------------------------------------------------
+class HasAttribute(HasAttributeBase):
+    """Rule that checks for a media object with a particular attribute"""
```
Comment (sam-m888) [inline gramps/gen/filters/rules/citation/_hasattribute.py]: > """Rule that checks for a media object with a particular attribute"""

Copy/paste error, should the doc strings in all four be identical?

Comment (Mattkmmr) [reply gramps/gen/filters/rules/citation/_hasattribute.py]: No that is a copy/paste error.

**Comment by prculley** [comment]
As of this moment, Places don't have attributes, so an attempt to use that filter should result in a crash.
If/when GEPS045 is accepted, places will have attributes...

**Comment by Mattkmmr** [comment]
@prculley will you add 'HasAttribute' for places with GEPS045 or should I put it on my ToDo list?

**Comment by prculley** [comment]
> 
> 
> @prculley will you add 'HasAttribute' for places with GEPS045 or should I put it on my ToDo list?

I already have it in my code base (but not pushed lately).

**Pull request merged**